lib/typed/success.rb

```diff
@@ -58,6 +58,15 @@ def and_then(&block)
       block.call(payload)
     end
 
+    sig do
+      override
+        .params(blk: T.proc.params(arg0: Error).void)
+        .returns(T.self_type)
+    end
+    def on_error(&blk) # rubocop:disable Lint/UnusedMethodArgument
```

Comment:
nit: you can name this `&_blk` to silence the rubocop rule and make it more explicit that the parameter won't be used

Reply:
Ahh right, because it's not a keyword. Thanks!